containerd-shim/process_solaris.go

```diff
@@ -0,0 +1,63 @@
+// +build solaris
+
+package main
+
+import (
+	"io"
+	"os"
+	"syscall"
+)
+
+// setPDeathSig is a no-op on Solaris as Pdeathsig is not defined.
+func setPDeathSig() *syscall.SysProcAttr {
+	return nil
+}
+
+// openIO opens the pre-created fifo's for use with the container
+// in RDWR so that they remain open if the other side stops listening
+func (p *process) openIO() error {
```

Comment:
This should be updated to use the `fifo` pkg from @tonistiigi  (see [here](https://github.com/docker/containerd/pull/330))